File: OBI - Treinamento/Fase 1/2023/contas.java
import java.util.Arrays;
import java.util.Scanner;

public class contas {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int v = scan.nextInt();
        int[] contas = new int[3];

        for (int i = 0; i < contas.length; i++) {
            contas[i] = scan.nextInt();
        }

        Arrays.sort(contas);

        int max = 0;
        for (int conta : contas) {
            if (v - conta < 0) {
                break;
            }
            v -= conta;
            max++;
        }
        System.out.println(max);
    }
}

File: OBI - Treinamento/Fase 1/2023/contas.cpp
//
// Created by xnaxg on 01/06/2025.
// Exercício: https://neps.academy/br/exercise/2321
// Resolvido em: 08m:33s
// Tempo total de prova (OBI 2023): 56m:32s
//

#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int v;
    vector<int> contas(3);
    cin >> v >> contas[0] >> contas[1] >> contas[2];

    sort(contas.begin(), contas.end());

    int pagas = 0;
    for (int conta : contas) {
        v -= conta;
        if (v < 0) continue;
        pagas++;
    }

    cout << pagas;
}
